src/store/useShapeDiverStoreParameters.ts

```diff
@@ -69,26 +71,59 @@ function createParameterExecutor<T>(sessionId: string, param: IGenericParameterD
 	};
 }
 
-function createGenericParameterExecutorForSession(session: ISessionApi) : IGenericParameterExecutor { 
+type DefaultExportsGetter = () => string[];
+type ExportResponseSetter = (response: IExportResponse) => void;
+
+function isFileParameter(param: IParameterApi<unknown>): param is IFileParameterApi {
+	return ("upload" in param);
+}
+
+function createGenericParameterExecutorForSession(session: ISessionApi, 
+	getDefaultExports: DefaultExportsGetter, exportResponseSetter: ExportResponseSetter) : IGenericParameterExecutor { 
 	
-	return (values) => {
+	return async (values) => {
 
 		// store previous values (we restore them in case of error)
 		const previousValues = Object.keys(values).reduce((acc, paramId) => {
 			acc[paramId] = session.parameters[paramId].value;
 			
 			return acc;
 		}, {} as { [paramId: string]: unknown});
+
+		// get ids of default exports that should be requested
+		const exports = getDefaultExports();
+
 		try {
 			// set values and call customize
 			Object.keys(values).forEach(id => session.parameters[id].value = values[id]);
-
-			return session.customize();
-		} 
+		
+			if (exports.length > 0) {
+				// prepare parameters (e.g. upload)
+				await Promise.all(Object.values(session.parameters)
+					.filter(isFileParameter)
+					.map(async p => {
+						const fileId = await p.upload();
+						p.value = fileId;
+					})
+				);
+				// prepare body and send request
+				const body: ShapeDiverRequestExport = { 
+					parameters: session.parameterValues as ShapeDiverRequestCustomization, // TODO fix this
```

Comment:
@MajorMeerkatThe3rd this was a little cumbersome to implement, because one has to take care of file uploads manually. Let's think about how we can improve this. As an example, it would be nice to be able to get the parameter values with a type that matches what's expected in `ShapeDiverRequestExport`. See [ISessionApi.parameterValues](https://viewer.shapediver.com/v3/latest/api/interfaces/ISessionApi.html#parameterValues)

Reply:
Related task: https://shapediver.atlassian.net/browse/SS-7487